python/triton/runtime/jit.py

```diff
@@ -319,15 +319,16 @@ def _make_launcher(self):
         args_signature = ', '.join(name if dflt == inspect._empty else f'{name} = {dflt}' for name, dflt in zip(self.arg_names, self.arg_defaults))
 
         src = f"""
-def {self.fn.__name__}({args_signature}, grid=None, num_warps=4, num_stages=3, extern_libs=None, stream=None, warmup=False, device=None, device_type=None):
+def {self.fn.__name__}({args_signature}, grid=None, extern_libs=None, stream=None, warmup=False, device=None, device_type=None, **kwargs):
     from ..compiler import compile, CompiledKernel
     sig_key =  {sig_keys},
     constexpr_key = {f'{constexpr_keys},' if len(constexpr_keys) > 0 else ()}
     spec_key = {f'{spec_keys},' if len(spec_keys) > 0 else ()}
+    num_warps = kwargs.get("num_warps")
+    num_stages = kwargs.get("num_stages")
     key = (version_key, sig_key, constexpr_key, spec_key, num_warps, num_stages, self.debug)
```

Comment:
For example, what're `num_warps` and `num_stages` here?

Reply:
If the code,
```
kernel[grid](..., num_warps=4);
```
Then it respects user's configuration and uses the `4` as part of the hash key and then pass down to the triton.compile to generate the kernel with num_warps=4.

If the code,
```
kernel[grid](...,);
```
Then the JITFunction uses the `None` as part of the hash key and let the triton.compile to choose the default value of the num_warps to generate the kernel.

Reply:
hmm, my point is that this PR will set `key`'s num_warps with a none value, but `key` will be used before compile right? More specifically, `key` is used to check `bin = cache[device].get(key, None)`